### PR TITLE
[docs] Add ciphertext validity proof docs

### PR DESCRIPTION
**Pull request by samkim-crypto**
#### Problem
The ciphertext validity proof latex docs have not been added to the ZK ElGamal Proof program docs page.

#### Summary of Changes
Wrote the security proof for ciphertext validity proofs and added it to the docs page.

Note for the reviewer: no need to go through the latex docs!

Fixes #

### CODE REVIEW

File: docs/src/runtime/zk-elgamal-proof.md [hunk 1]
```diff
@@ -122,3 +122,12 @@ proofs.
...
     [[Notes]](https://github.com/solana-labs/solana/blob/master/docs/src/runtime/zk-docs/ciphertext_ciphertext_equality.pdf)
+
+#### Ciphertext Validity proofs
+
+- `VerifyGroupedCiphertextValidity`:
+
+  - The grouped ciphertext validity proof certifies that a grouped ElGamal
+    cipehrtext is well-formed
+    - Mathematical description and proof of security:
+      [[Notes]](https://github.com/solana-labs/solana/blob/master/docs/src/runtime/zk-docs/ciphertext_validity.pdf)
```
Comment (joncinque) [inline docs/src/runtime/zk-elgamal-proof.md]: Should this point to agave sources? I don't know if this change will immediately get propagated to the solana repo

**Review by joncinque — COMMENTED** [review]
Looks good, just want to make sure the link goes to the right place

File: docs/src/runtime/zk-elgamal-proof.md (hunk 1 above)
Comment (samkim-crypto) [reply docs/src/runtime/zk-elgamal-proof.md]: Oh yes, I guess I should update the rest of the links as well.

**Review by joncinque — APPROVED** [review]
Looks great, thanks!

**Pull request merged**